CardPayments/src/main/java/com/paypal/android/cardpayments/VaultListener.kt

```diff
@@ -0,0 +1,15 @@
+package com.paypal.android.cardpayments
+
+import androidx.annotation.MainThread
+
+/**
+ * Listener to receive vault results.
+ */
+interface VaultListener {
+
+    /**
+     * Called after a successful vault occurs.
+     */
+    @MainThread
+    fun onVaultSuccess(result: VaultResult)
```

Comment:
Are we planning to add more cases, like if the attaching of the payment source to the setup token fails? Or if the user cancels the 3DS screen?

(looking at Jose's [doc](https://hackmd.io/2SUQKJh5TyGY_IozR4QUMQ?view#Kotlin))

Reply:
Yup! We hit a snag though, vault without purchase is more involved than we anticipated. At the moment, most of the request/response classes are placeholders. The bulk of this PR is UI / UX cleanup to get the vault feature flow into the demo app.

Reply:
Do we still have a vault request in our SDK?